### PR TITLE
issue #184 websocket timeout

### PR DESCRIPTION
**Pull request by albertwang-ibm**
Signed-off-by: Albert Wang <xuwang@us.ibm.com>

### CODE REVIEW

**Comment by albertwang-ibm** [comment]
The fix has been tested with timeout from 5 to 1 seconds, all work well.

File: fhir-notification/src/main/java/com/ibm/fhir/notification/util/FHIRNotificationUtil.java [hunk 1]
```diff
@@ -9,25 +9,34 @@
...
+import com.ibm.fhir.model.util.JsonSupport;
 import com.ibm.fhir.notification.FHIRNotificationEvent;
 
 public class FHIRNotificationUtil {
+    private static final JsonReaderFactory JSON_READER_FACTORY = Json.createReaderFactory(null);
+    private static final JsonBuilderFactory JSON_BUILDER_FACTORY = Json.createBuilderFactory(null);
     public static FHIRNotificationEvent toNotificationEvent(String jsonString) {
-        JsonReader reader = Json.createReader(new StringReader(jsonString));
+        try (JsonReader reader = JSON_READER_FACTORY.createReader(new StringReader(jsonString))) {
         JsonObject jsonObject = reader.readObject();
```
Comment (lmsurpre) [inline fhir-notification/src/main/java/com/ibm/fhir/notification/util/FHIRNotificationUtil.java]: shouldn't this block be indented now?

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRNotificationServiceClientEndpoint.java [hunk 2]
```diff
@@ -53,8 +53,8 @@ public void onOpen(Session session, EndpointConfig config) {
         session.addMessageHandler(new MessageHandler.Whole<String>() {
             public void onMessage(String text) {
                 System.out.println(">>> Received message: " + text);
-                FHIRNotificationEvent event = FHIRNotificationUtil.toNotificationEvent(text);
-                events.add(event);
+            //    FHIRNotificationEvent event = FHIRNotificationUtil.toNotificationEvent(text);
```
Comment (lmsurpre) [inline fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRNotificationServiceClientEndpoint.java]: why is this line commented out?  is it safe to remove?

File: fhir-notification/src/main/java/com/ibm/fhir/notification/util/FHIRNotificationUtil.java (hunk 1 above)
Comment (albertwang-ibm) [reply fhir-notification/src/main/java/com/ibm/fhir/notification/util/FHIRNotificationUtil.java]: ha, good eye. yes.

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRNotificationServiceClientEndpoint.java (hunk 2 above)
Comment (albertwang-ibm) [reply fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRNotificationServiceClientEndpoint.java]: safe to remove, I was wondering if I should add comments about reason for the change. let me remove

File: fhir-notification/src/main/java/com/ibm/fhir/notification/util/FHIRNotificationUtil.java (hunk 1 above)
Comment (albertwang-ibm) [reply fhir-notification/src/main/java/com/ibm/fhir/notification/util/FHIRNotificationUtil.java]: done

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRNotificationServiceClientEndpoint.java (hunk 2 above)
Comment (albertwang-ibm) [reply fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRNotificationServiceClientEndpoint.java]: done

**Comment by prb112** [comment]
interesting... makes sense

**Pull request merged**